src/cryocanvas/app.py

```diff
@@ -151,143 +157,167 @@ def get_prediction_layer(self):
     def get_painting_layer(self):
         return self.viewer.layers["Painting"]
 
-    def on_data_change(self, event, app):
-        # Define corner_pixels based on the current view or other logic
-        corner_pixels = self.viewer.layers["Image"].corner_pixels
-
-        # Start the thread with correct arguments
-        thread = threading.Thread(
-            target=self.threaded_on_data_change,
-            args=(
-                event,
-                corner_pixels,
-                self.viewer.dims.current_step,
-                self.widget.model_dropdown.currentText(),
-                {
-                    "basic": self.widget.basic_checkbox.isChecked(),
-                    "embedding": self.widget.embedding_checkbox.isChecked(),
-                },
-                self.widget.live_fit_checkbox.isChecked(),
-                self.widget.live_pred_checkbox.isChecked(),
-                self.widget.data_dropdown.currentText(),
-            ),
-        )
-        thread.start()
-        thread.join()
+    @ensure_main_thread
```

Comment:
The docs of `ensure_main_thread` say that the function passed (or decorated as here) should be the method of a `QObject`. This is not the case here or on `main`. But after digging through the code quickly, I'm not sure why as `ensure_main_thread` creates a wrapping `QObject` that may guarantee what's needed to actually ensure this runs on Qt's main thread.